DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/hostpipes/src/hostpipes.cpp

```diff
@@ -177,6 +177,7 @@ void LaunchCollectTest(sycl::queue& q, ValueT* in, ValueT* out, size_t count,
                        size_t repeats) {
   std::cout << "\t Run Loopback Kernel on FPGA" << std::endl;
 
+#if defined(FPGA_SIMULATOR) || defined(FPGA_HARDWARE)
```

Comment:
at a glance the code in these blocks appears identical. please only use the compiler macros for minimal spans of code so it's clear to the reader what is different between emulation and simulation. Ideally, there should be **difference**, as forcing users to write different code for emulation and simulation defeats the point of using high-level design.

Reply:
Updated, thanks.